addons/account/models/account_move.py

```diff
@@ -1900,6 +1900,24 @@ def post(self):
                 # installing Accounting- with bank statements)
                 move.company_id.account_bank_reconciliation_start = move.date
 
+        for move in self:
+            if not move.partner_id: continue
+            if move.type.startswith('out_'):
+                field='customer_rank'
+            elif move.type.startswith('in_'):
+                field='supplier_rank'
+            else:
+                continue
+            try:
+                with self.env.cr.savepoint():
+                    self.env.cr.execute("UPDATE res_partner SET "+field+"="+field+"+1 WHERE ID=%s FOR UPDATE NOWAIT", (move.partner_id.id,))
```

Comment:
`self.env.cr.execute("UPDATE res_partner SET "{field}"="{field}"+1 WHERE ID=%s FOR UPDATE NOWAIT".format(field=field), (move.partner_id.id,))`